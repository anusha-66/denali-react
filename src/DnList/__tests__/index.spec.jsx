/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnList } from '../../index'; // src/index.tsx

describe('DnList Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnList>
                <DnList.Item>List item 1</DnList.Item>
                <DnList.Item>List item 2</DnList.Item>
                <DnList.Item>List item 3</DnList.Item>
                <DnList.Item>List item 4</DnList.Item>
            </DnList>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnList isOrderedList className="dummy-list-classname">
                <DnList.Item className="dummy-list-item-classname">List item 1</DnList.Item>
                <DnList.Item>List item 2</DnList.Item>
                <DnList.Item>List item 3</DnList.Item>
                <DnList.Item>List item 4</DnList.Item>
            </DnList>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
