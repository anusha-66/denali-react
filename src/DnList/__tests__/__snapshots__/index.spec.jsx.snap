// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnList Tests should render correctly with custom props 1`] = `
<ol
  class="list dummy-list-classname"
>
  <li
    class="dummy-list-item-classname"
  >
    List item 1
  </li>
  <li>
    List item 2
  </li>
  <li>
    List item 3
  </li>
  <li>
    List item 4
  </li>
</ol>
`;

exports[`DnList Tests should render with default props 1`] = `
<ul
  class="list"
>
  <li>
    List item 1
  </li>
  <li>
    List item 2
  </li>
  <li>
    List item 3
  </li>
  <li>
    List item 4
  </li>
</ul>
`;
